src/firestore/src/convert.rs

```diff
@@ -0,0 +1,227 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains integration tests for the generated `Convert` implementations.
+//!
+//! For generated code, we write hand-crafted integration tests *once* in the
+//! project. The generated implementations for `wkt::prost::Convert` usually
+//! reference a type generated by Protobuf, and these are `pub(crate)`.
+//! Therefore, the tests must be part of the crate that contains them, and
+//! cannot be a separate integration tests crate, or a file in the `tests/`
+//! directory.
+//!
+//! The strategy we follow is to write one test per category of generated code.
+//! Exhaustive tests of each generated code quickly get repetitive. The most
+//! sources of variation in these implementations are the type of fields. Such
+//! as:
+//!
+//! - Simple primitive fields
+//! - Enum fields
+//! - Repeated fields
+//! - Map fields
+//! - OneOf fields
+//! - Optional fields
+//!
+//! And then you need to consider variations on these, such as oneofs that
+//! contain enums, or maps that contain enums.
+
+#[cfg(test)]
+mod test {
+    use crate::generated::model;
+    use crate::google;
+    use wkt::prost::Convert;
+
+    #[test]
+    fn test_basic_fields() {
+        let sidekick = gtype::model::LatLng::new()
+            .set_latitude(12.5)
+            .set_longitude(34.5);
+        let proto = google::r#type::LatLng {
+            latitude: 12.5,
+            longitude: 34.5,
+        };
+
+        let got = sidekick.clone().cnv();
+        assert_eq!(got, proto);
+
+        let got = proto.clone().cnv();
+        assert_eq!(got, sidekick);
+    }
+
+    #[test]
+    fn test_enum_field() {
+        let sidekick = model::structured_query::FieldFilter::new()
+            .set_op(model::structured_query::field_filter::Operator::EQUAL);
+        let proto = google::firestore::v1::structured_query::FieldFilter {
+            op: google::firestore::v1::structured_query::field_filter::Operator::Equal.into(),
+            ..Default::default()
+        };
+
+        let got = sidekick.clone().cnv();
+        assert_eq!(got, proto);
+
+        let got = proto.clone().cnv();
+        assert_eq!(got, sidekick);
+    }
+
+    #[test]
+    fn test_optional_field() {
+        let sidekick = model::structured_query::UnaryFilter::new().set_operand_type(
+            model::structured_query::unary_filter::OperandType::Field(Box::new(
+                model::structured_query::FieldReference::new().set_field_path("a.b.c"),
+            )),
+        );
+        let proto = google::firestore::v1::structured_query::UnaryFilter {
+            operand_type:
+                google::firestore::v1::structured_query::unary_filter::OperandType::Field(
+                    google::firestore::v1::structured_query::FieldReference {
+                        field_path: "a.b.c".into(),
+                        ..Default::default()
```

Comment:
(so I can learn) how do I read `..Default::default()`? Like, what do the leading dots mean?

Reply:
I have been cargo culting this syntax without reading what it means. I guess when I read this chapter the idea did not sink in:

https://doc.rust-lang.org/stable/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax

Reply:
Ah, that is cool, thanks. TIL: Struct update syntax. This could have simplified a lot of my auth unit tests.

Also TIL: cargo culting. Unrelated to `cargo`.